### PR TITLE
SwinIR Support

### PR DESCRIPTION
**Pull request by joeyballentine**
- Adds SwinIR support for all models
- The gray models are currently broken, but will be fixed by #721
- The "classical sr" models are unable to use fp16, hence the new flags in each model. However, they are able to use bfloat16, so we should add support for that, which is why I suggested #811 
- I also switched from manually casting the model and image to half/float to just using the autocast context manager pytorch provides. This ensures that only operations that are actually able to be cast to fp16 become fp16. Unfortunately, this did not fix the issue with the "classical sr" models, but it is still good to have.

### CODE REVIEW

File: backend/src/nodes/properties/outputs/pytorch_outputs.py [hunk 1]
```diff
@@ -21,6 +21,19 @@ def get_broadcast_data(self, value: PyTorchModel):
...
         if "SRVGG" in value.model_type:  # type: ignore
             size = [f"{value.num_feat}nf", f"{value.num_conv}nc"]
+        elif "SwinIR" in value.model_type:  # type: ignore
+            head_length = len(value.depths)  # type: ignore
+            if head_length <= 4:
+                size_tag = "small"
+            elif head_length < 9:
+                size_tag = "medium"
+            else:
+                size_tag = "large"
```
Comment (RunDevelopment) [inline backend/src/nodes/properties/outputs/pytorch_outputs.py:31]: This seems arbitrary. Could you explain this?

**Review by RunDevelopment — COMMENTED** [review]
Lots and lots of additions in this PR. Frankly, all that stuff is over my head anyway, so I'm just gonna say that it's probably fine.

File: backend/src/nodes/properties/outputs/pytorch_outputs.py (hunk 1 above)
Comment (joeyballentine) [reply backend/src/nodes/properties/outputs/pytorch_outputs.py:31]: https://github.com/JingyunLiang/SwinIR/releases/tag/v0.0

The S, M and L in the model names

Comment (RunDevelopment) [reply backend/src/nodes/properties/outputs/pytorch_outputs.py:31]: I guess it's fine then.

**Comment by joeyballentine** [comment]
Just realized I need to update the readme with this

**Review by RunDevelopment — APPROVED** [review]
Should be fine.

**Pull request merged**